google/cloud/aiplatform/prediction/model_server.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from typing import Type
+
+try:
+    from fastapi import FastAPI
+    from fastapi import Request
+    from fastapi import Response
+except ImportError:
+    raise ImportError(
+        "FastAPI is not installed and is required to build model servers. "
+        'Please install the SDK using "pip install python-aiplatform[prediction]"'
+    )
+
+try:
+    import uvicorn
+except ImportError:
+    raise ImportError(
+        "Uvicorn is not installed and is required to run fastapi application. "
+        'Please install the SDK using "pip install python-aiplatform[prediction]"'
+    )
+
+from google.cloud.aiplatform.prediction.handler import DefaultHandler
+from google.cloud.aiplatform.prediction.handler import Handler
+from google.cloud.aiplatform.prediction.predictor import Predictor
+
+
+class ModelServer:
+    """Base model server to do custom prediction routines."""
+
+    def __init__(
+        self, predictor: Predictor, handler_class: Type[Handler] = DefaultHandler
```

Comment:
This will only require a `Handler`, and the `ModelServer` will only have a concept of the `Handler`. 

Under the hood, the `Handler` may use a `Predictor`, but we won't require it. The `Handler` will be responsible for interacting with the `Predictor`.

Reply:
We can let `ModelServer` to have `Handler` only. Then we will put `handler` initialization before we pass it to `ModelServer` (the time that `ModelServer` initialization is in entrypoint.py now). And the time we call `Predictor.load` will be in `handler` initialization.

Reply:
Updated the code. PTAL.